pkg/arch/hook.go

```diff
@@ -110,13 +110,14 @@ func (w warning) Error() string {
 
 type HandlerOption func(*Handler)
 type Handler struct {
-	Client                client.Client
-	Registry              Registry
-	matchNodeLabels       []string
-	metrics               HandlerMetrics
-	decoder               *admission.Decoder
-	preferredArchitecture string
-	systemOS              string
+	Client                   client.Client
```

Comment:
nit: maybe creating a HandlerOpts struct?

Reply:
You prefer a HandlerOpts struct instead of the HandlerOption functions approach? The resulting public API should be almost equally ergonomic in both cases. What do you envision exactly?

Reply:
i think they are equally ergonomic indeed so no need for this